### PR TITLE
PMM-12624 Single inventory service add endpoint

### PR DESCRIPTION
**Pull request by ademidoff**
PMM-12624

Link to the Feature Build: SUBMODULES-3442

- [x] API Docs updated

Screenshot 2023-11-10 at 13 43 29

### CODE REVIEW

**Review by idoqo — APPROVED** [review]

File: managed/services/inventory/grpc/services_server.go [hunk 1]
```diff
@@ -140,8 +140,52 @@ func (s *servicesServer) GetService(ctx context.Context, req *inventorypb.GetSer
...
-func (s *servicesServer) AddMySQLService(ctx context.Context, req *inventorypb.AddMySQLServiceRequest) (*inventorypb.AddMySQLServiceResponse, error) {
+// AddService adds any type of Service.
+func (s *servicesServer) AddService(ctx context.Context, req *inventorypb.AddServiceRequest) (*inventorypb.AddServiceResponse, error) {
+	switch req.Service.(type) {
+	case *inventorypb.AddServiceRequest_Mysql:
+		resp, err := s.addMySQLService(ctx, req.GetMysql())
+		if err != nil {
+			return nil, err
+		}
+		return resp, nil
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply managed/services/inventory/grpc/services_server.go]: Sure, thanks!

**Review by artemgavrilov — APPROVED** [review]
LGTM

File: api/inventorypb/services.proto [hunk 2]
```diff
@@ -214,8 +214,30 @@ message GetServiceResponse {
...
+}
+
+message AddServiceRequest {
+  oneof service {
+    AddMySQLServiceRequest mysql = 1;
+    AddMongoDBServiceRequest mongodb = 2;
+    AddPostgreSQLServiceRequest postgresql = 3;
+    AddProxySQLServiceRequest proxysql = 4;
+    AddHAProxyServiceRequest haproxy = 5;
+    AddExternalServiceRequest external = 6;
```
[the remark this answers is not in the document]
Comment (ademidoff) [reply api/inventorypb/services.proto]: It's a good suggestion, thanks! Done.

File: api-tests/inventory/helpers.go [hunk 3]
```diff
@@ -91,79 +91,14 @@ func addAzureDatabaseExporter(t pmmapitests.TestingT, body agents.AddAzureDataba
...
-}
-
-func addHAProxyService(t pmmapitests.TestingT, body services.AddHAProxyServiceBody) *services.AddHAProxyServiceOKBody {
-	t.Helper()
-
-	params := &services.AddHAProxyServiceParams{
-		Body:    body,
-		Context: pmmapitests.Context,
-	}
-	res, err := client.Default.Services.AddHAProxyService(params)
```
Comment (BupycHuk) [inline api-tests/inventory/helpers.go:166]: great job!

File: docs/api/inventory/add-node.md [hunk 4]
```diff
@@ -6,16 +6,14 @@ category: 626de009b977e3003179f7dd
 
 ## Add a Node
 
-This section describes how to add a Node of any type to the inventory.
+This section describes how to add a Node of any type to PMM Inventory.
 
-In PMM versions prior to 2.40.0, we featured a separate API call for each Node type. Starting with PMM 2.40.0, we have a single API call for all Node types. The API call is `Add` and the Node type is specified in the `node_type` field. The `node_type` field is required. Along with this single API endpoint, we are deprecating the separate API calls for each Node type.
+In PMM versions prior to 2.40.0, we featured a separate API call for each Node type. Starting with PMM 2.40.0, we offer a single API endpoint for all Node types. While previously the Node type was defined by the endpoint, i.e. `Nodes/AddGeneric`, now the Node type must be specified as the top-level property of the request payload. Along with this single API endpoint, we are deprecating the separate API calls for each Node type.
```
Comment (BupycHuk) [inline docs/api/inventory/add-node.md:11]:
```suggestion
In PMM 2, we featured a separate API call for each Node type. Starting with PMM 3, we offer a single API endpoint for all Node types. While previously the Node type was defined by the endpoint, i.e. `Nodes/AddGeneric`, now the Node type must be specified as the top-level property of the request payload. Along with this single API endpoint, we are deprecating the separate API calls for each Node type.
```

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**